{
  "WorkItem": {
    "AffectedComponent": {
      "Name": "",
      "DisplayName": ""
    },
    "ClosedComment": "In the latest version this compiles correctly without error.",
    "ClosedDate": "2014-03-03T12:00:18.93-08:00",
    "CommentCount": 0,
    "Custom": null,
    "Description": "This small code snippet: \r\n\r\n```\r\n// Aux functions to help the class... can also be done as static private in\r\n// the class, but problem: sometime more convenient to have in a separate file.\r\nfunction FooHelper(x: Foo.Request) {\r\n  // ... do helpful stuff...\r\n}\r\n\r\n// Interface for requesting a new foo object.\r\nclass Foo {\r\n  constructor (public x: Foo.Request) {\r\n    FooHelper(x);\r\n  }\r\n  //... other stuff for Foo...\r\n}\r\n\r\n// static module addition of\r\nmodule Foo {\r\n  export interface Request {\r\n    a: string;\r\n    b: string;\r\n  }\r\n\r\n  // helper stuff you don't want in the public\r\n  // ...\r\n  // Export helper function, e.g.:\r\n  export function makeTrivialFoo(x:string) { return new Foo({a: x, b:x}); }\r\n}\r\n\r\nvar foo = Foo.makeTrivialFoo(\"a\");\r\n```\r\n\r\nproduces this error: \r\n\r\n```\r\n/Users/ldixon/code/safepassage/UProxy/scraps/ts3.ts(28,15): error TS2094: The property 'makeTrivialFoo' does not exist on value of type 'typeof Foo'.\r\n```\r\n\r\nThe error is confused. By commenting out the helper function the error does not happen. i.e. the following is fine: \r\n\r\n```\r\n// Aux functions to help the class... can also be done as static private in\r\n// the class, but problem: sometime more convenient to have in a separate file.\r\n// Interface for requesting a new foo object.\r\nclass Foo {\r\n  constructor (public x: Foo.Request) {\r\n  }\r\n  //... other stuff for Foo...\r\n}\r\n\r\n// static module addition of\r\nmodule Foo {\r\n  export interface Request {\r\n    a: string;\r\n    b: string;\r\n  }\r\n\r\n  // helper stuff you don't want in the public\r\n  // ...\r\n  // Export helper function, e.g.:\r\n  export function makeTrivialFoo(x:string) { return new Foo({a: x, b:x}); }\r\n}\r\n\r\nvar foo = Foo.makeTrivialFoo(\"a\");\r\n```\r\n\r\nI don't think there should have even been an error for the first case. \r\n\r\n\r\nUsing: tsc Version 0.9.5.0",
    "LastUpdatedDate": "2014-03-03T12:00:18.93-08:00",
    "PlannedForRelease": "",
    "ReleaseVisibleToPublic": false,
    "Priority": {
      "Name": "Unassigned",
      "Severity": 0,
      "Id": 0
    },
    "ProjectName": "typescript",
    "ReportedDate": "2014-03-02T15:24:22.53-08:00",
    "Status": {
      "Name": "Closed",
      "Id": 4
    },
    "ReasonClosed": {
      "Name": "Fixed"
    },
    "Summary": "Confused (wrong) error for module after class definition",
    "Type": {
      "Name": "Unassigned",
      "Id": 5
    },
    "VoteCount": 1,
    "Id": 2251
  },
  "FileAttachments": [],
  "Comments": []
}